src/Core/Grid/Query/ContactsQueryBuilder.php

```diff
@@ -0,0 +1,138 @@
+ */
+final class ContactsQueryBuilder extends AbstractDoctrineQueryBuilder
+{
+    /**
+     * @var DoctrineSearchCriteriaApplicatorInterface
+     */
+    private $searchCriteriaApplicator;
+
+    /**
+     * @var int
+     */
+    private $languageId;
+
+    /**
+     * @var array
+     */
+    private $contextShopsIds;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator
+     * @param int $languageId
+     * @param array $contextShopsIds
+     */
+    public function __construct(
+        Connection $connection,
+        $dbPrefix,
+        DoctrineSearchCriteriaApplicatorInterface $searchCriteriaApplicator,
+        $languageId,
+        array $contextShopsIds
+    ) {
+        parent::__construct($connection, $dbPrefix);
+
+        $this->searchCriteriaApplicator = $searchCriteriaApplicator;
+        $this->languageId = $languageId;
+        $this->contextShopsIds = $contextShopsIds;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSearchQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters());
+        $qb->select('c.id_contact, c.email, cl.name, cl.description');
+
+        $this->searchCriteriaApplicator
+            ->applySorting($searchCriteria, $qb)
+            ->applyPagination($searchCriteria, $qb)
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCountQueryBuilder(SearchCriteriaInterface $searchCriteria)
+    {
+        $qb = $this->getQueryBuilder($searchCriteria->getFilters())
+            ->select('SQL_CALC_FOUND_ROWS')
+            ->select('FOUND_ROWS()')
+        ;
+
+        return $qb;
+    }
+
+    /**
+     * Get generic query builder.
+     *
+     * @param array $filters
+     *
+     * @return QueryBuilder
+     */
+    private function getQueryBuilder(array $filters)
+    {
+        $qb = $this->connection
+            ->createQueryBuilder()
+            ->from($this->dbPrefix . 'contact', 'c')
+            ->innerJoin('c', $this->dbPrefix . 'contact_lang', 'cl', 'c.id_contact = cl.id_contact')
+            ->innerJoin('c', $this->dbPrefix . 'contact_shop', 'cs', 'c.id_contact = cs.id_contact')
+            ->andWhere('cl.`id_lang`= :language')
+            ->andWhere('cs.`id_shop` IN (:shops)')
+            ->setParameter('language', $this->languageId)
+            ->setParameter('shops', $this->contextShopsIds, Connection::PARAM_INT_ARRAY)
+            ->groupBy('c.id_contact')
+        ;
+
+        foreach ($filters as $name => $value) {
+            if ('id_contact' === $name) {
+                $qb->andWhere('c.`id_contact` = :' . $name);
+                $qb->setParameter($name, $value);
+
+                continue;
+            }
+
+            $qb->andWhere("$name LIKE :$name");
```

Comment:
Although it is unlikely, I am not confortable with "$..." syntax for security reasons (SQL injections) :/
Can you protect this field using backquotes "`" ?

Reply:
using backquotes not completlty protect against sqli here. more if the `$name` contains a string like this: 
```
name` OR 1=1 OR `name
```

The secure way is to define each field you want to use, or check if these fields exist in ObjectModel

Reply:
@PierreRambaud it would solve (possible) security issue, but add limitation. Dont forget that filters are dynamic, they can (and will) be extended/modified by modules, so checking on object model would not work. :/

Reply:
ok, i think i see a possible improvement for it. But at the moment all filters (default & from modules) are handled here.

Reply:
We found an issue to solve in 1.7.6 here 😃 ! "What is the proper way to handle search if a grid is extended ?"
Proper = easy, documented and reliable of course 😅

Reply:
Issue created: https://github.com/PrestaShop/PrestaShop/issues/10823

Reply:
i think that all we have to do is pass `SearchCriteria` to the hook so modules can handle their filters however they want to, wdyt?

Reply:
We'll see in the issue (I have copied your suggestion there)